src/Microsoft.Dnx.Compilation.CSharp/RoslynCompiler.cs

```diff
@@ -286,17 +286,22 @@ private static string RemovePrereleaseTag(string version)
             // Simple reparse of the version string (because we don't want to pull in NuGet stuff
             // here because we're in an old-runtime/new-runtime limbo)
 
-            var dashIdx = version.IndexOf('-');
+            var dashIdx = version?.IndexOf('-');
             if (dashIdx < 0)
             {
                 return version;
             }
             else
             {
-                return version.Substring(0, dashIdx);
+                return version?.Substring(0, dashIdx.Value);
             }
         }
 
+        private static string EscapeCharacters(string str)
+        {
+            return str != null ? SymbolDisplay.FormatLiteral(str, false) : null;
```

Comment:
http://www.coderesx.com/roslyn/html/D316757D.htm

"Escapes non-printable characters." <- is this what you want?